### PR TITLE
[CN-276] Configuration of persistence feature for members

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: api/v1alpha1/hazelcast_types.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"hash/fnv"
 
+	"k8s.io/apimachinery/pkg/api/resource"
+
```
Comment (hakanmemisoglu) [inline api/v1alpha1/hazelcast_types.go:8]:
```suggestion
```

File: api/v1alpha1/hazelcast_types.go [hunk 2]
```diff
@@ -59,8 +61,66 @@ type HazelcastSpec struct {
...
+	// Configuration of PersistenceVolumeClaim.
+	Pvc PersistencePvcConfiguration `json:"pvc"`
+}
+
+type PersistencePvcConfiguration struct {
+	// Name of the existing PVC.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// AccessModes contains the actual access modes the volume backing the PVC has.
```
Comment (hakanmemisoglu) [inline api/v1alpha1/hazelcast_types.go:90]:
```suggestion
	// AccessModes contains the actual access modes of the volume backing the PVC has.
```

File: api/v1alpha1/hazelcast_types.go [hunk 3]
```diff
@@ -59,8 +61,66 @@ type HazelcastSpec struct {
...
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
+	// +optional
+	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes,omitempty"`
+
+	// A description of the PVC request capacity.
+	// +optional
+	RequestStorage resource.Quantity `json:"requestStorage,omitempty"`
+
+	// Name of StorageClass to which this persistent volume belongs.
```
Comment (hakanmemisoglu) [inline api/v1alpha1/hazelcast_types.go:99]:
```suggestion
	// Name of StorageClass which this persistent volume belongs to.
```

File: test/matcher.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"fmt"
+
+	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
```
Comment (hakanmemisoglu) [inline test/matcher.go:8]:
```suggestion
	"github.com/onsi/gomega/format"
	"github.com/onsi/gomega/types"
	
	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
```

File: test/spec.go [hunk 5]
```diff
@@ -1,9 +1,8 @@
 package test
 
 import (
-	. "github.com/onsi/gomega"
-
 	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 )
```
Comment (hakanmemisoglu) [inline test/spec.go:7]:
```suggestion
	. "github.com/onsi/gomega"
	corev1 "k8s.io/api/core/v1"
	
	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
)
```

File: test/integration/hazelcast_controller_test.go [hunk 6]
```diff
@@ -382,20 +382,20 @@ var _ = Describe("Hazelcast controller", func() {
...
-			ClusterSize:      n.DefaultClusterSize,
-			Repository:       n.HazelcastRepo,
-			Version:          n.HazelcastVersion,
-			LicenseKeySecret: "",
-			ImagePullPolicy:  n.HazelcastImagePullPolicy,
+		defaultHzSpecs := &test.HazelcastSpecValues{
+			ClusterSize:     n.DefaultClusterSize,
+			Repository:      n.HazelcastRepo,
+			Version:         n.HazelcastVersion,
+			LicenseKey:      "",
```
Comment (hakanmemisoglu) [inline test/integration/hazelcast_controller_test.go:389]: I think LicenseKeySecret is still better name for this field.

**Review by hakanmemisoglu — APPROVED** [review]
LGTM. I have just few comments on formatting, docs etc. Feel free to discard them. 
I will run it on real cluster tomorrow.

File: test/integration/hazelcast_controller_test.go (hunk 6 above)
Comment (SeriyBg) [reply test/integration/hazelcast_controller_test.go:389]: I didn't rename the field. I've just changed the `hazelcastv1alpha1.HazelcastSpec` to `test.HazelcastSpecValues`, which has a different field 😄

Comment (hakanmemisoglu) [reply test/integration/hazelcast_controller_test.go:389]: Oops!
Who introduced this field?
I found [him](https://github.com/hazelcast/hazelcast-platform-operator/blob/f6865d38580d039ab9334711797824ef3fa40064/test/spec.go).
😆

**Review by hasancelik — APPROVED** [review]

**Pull request merged**